test/network/2020-03-01/armnetwork/pollers.go

```diff
@@ -8867,120 +3548,47 @@ type StringPoller interface {
 type stringPoller struct {
 	// the client for making the request
 	pipeline azcore.Pipeline
-	pt       pollingTracker
+	pt       armcore.Poller
 }
 
 // Done returns true if there was an error or polling has reached a terminal state
 func (p *stringPoller) Done() bool {
-	return p.pt.hasTerminated()
+	return p.pt.Done()
 }
 
 // Poll will send poll the service endpoint and return an http.Response or error received from the service
 func (p *stringPoller) Poll(ctx context.Context) (*http.Response, error) {
-	if lroPollDone(ctx, p.pipeline, p.pt) {
-		return p.pt.latestResponse().Response, p.pt.pollingError()
-	}
-	return nil, p.pt.pollingError()
+	return p.pt.Poll(ctx, p.pipeline)
 }
 
 func (p *stringPoller) FinalResponse(ctx context.Context) (*StringResponse, error) {
-	if !p.Done() {
-		return nil, errors.New("cannot return a final response from a poller in a non-terminal state")
-	}
-	if p.pt.pollerMethodVerb() == http.MethodPut || p.pt.pollerMethodVerb() == http.MethodPatch {
-		res, err := p.handleResponse(p.pt.latestResponse())
-		if err != nil {
-			return nil, err
-		}
-		if res != nil && (*res.Value != "") {
-			return res, nil
-		}
-	}
-	// checking if there was a FinalStateVia configuration to re-route the final GET
-	// request to the value specified in the FinalStateVia property on the poller
-	err := p.pt.setFinalState()
-	if err != nil {
-		return nil, err
-	}
-	if p.pt.finalGetURL() == "" {
-		// we can end up in this situation if the async operation returns a 200
-		// with no polling URLs.  in that case return the response which should
-		// contain the JSON payload (only do this for successful terminal cases).
-		if lr := p.pt.latestResponse(); lr != nil && p.pt.hasSucceeded() {
-			result, err := p.handleResponse(lr)
-			if err != nil {
-				return nil, err
-			}
-			return result, nil
-		}
-		return nil, errors.New("missing URL for retrieving result")
-	}
-	u, err := url.Parse(p.pt.finalGetURL())
-	if err != nil {
-		return nil, err
-	}
-	req := azcore.NewRequest(http.MethodGet, *u)
-	if err != nil {
-		return nil, err
-	}
-	resp, err := p.pipeline.Do(ctx, req)
+	temp := ""
```

Comment:
@jhendrixMSFT look at this

Reply:
Yes I remember now.  I misspoke earlier, we don't need to populate `ResponseEnvelope.Widget` before handing it off to the unmarshaller, so we should be able to remove this.

Reply:
But we do for scalar types, because if not it's nil and then json.Unmarhsal fails since it's nil.

Reply:
Done